parquet/src/main/java/org/apache/iceberg/parquet/ColumnarBatchReader.java

```diff
@@ -45,9 +45,11 @@ public final ColumnarBatch read(ColumnarBatch ignore) {
 
             FieldVector vec = readers[i].read();
             arrowVectorArr[i] = new ArrowColumnVector(vec);
-            Preconditions.checkState(i > 0 && numRows == vec.getValueCount(),
+            if (i > 0) {
```

Comment:
Isn't this just `Preconditions.checkState(i > 0 || numRows == vec.getValueCount(), ...)`?